src/core/ai/agent/SaikiAgent.ts

```diff
@@ -210,42 +230,109 @@ export class SaikiAgent {
      * @throws Error if session doesn't exist
      */
     public async getSessionHistory(sessionId: string) {
-        const session = this.sessionManager.getSession(sessionId);
+        const session = await this.sessionManager.getSession(sessionId);
         if (!session) {
             throw new Error(`Session '${sessionId}' not found`);
         }
         return await session.getHistory();
     }
 
+    /**
+     * Loads a session as the new "default" session for this agent.
+     * All subsequent operations that don't specify a session ID will use this session.
+     * This provides a clean "current working session" pattern for API users.
+     *
+     * @param sessionId The session ID to load as default, or null to reset to original default
+     * @throws Error if session doesn't exist
+     *
+     * @example
+     * ```typescript
+     * // Load a specific session as default
+     * await agent.loadSession('project-alpha');
+     * await agent.run("What's the status?"); // Uses project-alpha session
+     *
+     * // Reset to original default
+     * await agent.loadSession(null);
+     * await agent.run("Hello"); // Uses 'default' session
+     * ```
+     */
+    public async loadSession(sessionId: string | null = null): Promise<void> {
```

Comment:
should return back a session object instead of this,

move this into a different method maybe `setDefaultSession`?